simtel_pipeline.py

```diff
@@ -0,0 +1,279 @@
+    # Each pixel in each event has its own baseline
+    #
+    # 'simtel'
+    # Baseline is taken as simulated value event.mc.pedestal/50 that can
+    # be set up with the use of variable 'fadc_pedestal'
+    # in CTA-ULTRA6-SST-DC.cfg.
+
+    data_stream = simtel_baseline.fill_baseline_r0(
+        data_stream, n_bins0=options.baseline_beginning,
+        n_bins1=options.baseline_end, method='data')
+
+    # Run the r1 calibration (i.e baseline substraction)
+    data_stream = r1.calibrate_to_r1(data_stream, dark_baseline)
+
+    # Run the dl1 calibration (compute charge in photons + cleaning)
+    data_stream = dl1.calibrate_to_dl1(
+        data_stream,
+        time_integration_options,
+        additional_mask=additional_mask,
+        picture_threshold=picture_threshold,
+        boundary_threshold=boundary_threshold)
+
+    # Return only showers with total number of p.e. above min_photon
+    data_stream = filter.filter_shower(data_stream, min_photon=min_photon)
+
+    # Suffix for output filenames
+    suffix = (
+              options.primary +
+              '_ze' + str(options.zenit_angle).zfill(2) +
+              '_az' + str(options.azimut).zfill(3) +
+              '_p' + str(options.picture_threshold).zfill(2) +
+              '_b' + str(options.boundary_threshold).zfill(2)
+             )
+
+    # Save cleaned events - pixels and corresponding values
+    filename_pix = 'pixels.txt'
+    filename_eventsimage = 'events_image_' + suffix + '.txt'
+    data_stream = events_image.save_events(
+        data_stream, directory + filename_pix,
+        directory + filename_eventsimage)
+
+    # Save simulated shower paramters
+    filename_showerparam = 'pipedmc_param_' + suffix + '.txt'
+    data_stream = mc_shower.save_shower(
+        data_stream, directory + filename_showerparam)
+
+    # Run the dl2 calibration (Hillas)
+    data_stream = dl2.calibrate_to_dl2(
+        data_stream, reclean=reclean,
+        shower_distance=shower_distance)
+
+    # Save arrival times of photons in pixels passed cleaning
+    filename_timing = 'timing_' + suffix + '.txt'
+    data_stream = events_image.save_timing(
+        data_stream, directory + filename_timing)
+
+    # Save mean baseline in event pixels
+    filename_baseline = (
+                         'baseline_' + suffix +
+                         '_bas' + str(options.baseline_beginning).zfill(2) +
+                         str(options.baseline_end).zfill(2) + '.txt'
+                         )
+    # data_stream = simtel_baseline.save_mean_event_baseline(
+    #    data_stream, directory + filename_baseline)
+
+    # Save Hillas
+    hillas_filename = options.output + '_' + suffix
+    save_hillas_parameters(
+        data_stream=data_stream,
+        n_showers=n_showers,
+        output_filename=directory + hillas_filename)
+    # save_hillas_parameters_in_text(
+    #    data_stream=data_stream,
+    #    output_filename=directory + hillas_filename)
+
```

Comment:
This is the main script of this PR. In the bottom is a lot of commented out plotting code. I find it great to split plotting from doing the analysis. So .. can this be removed?

pipeline_crab.py has evolved a bit, since this was written, I assume. So they look very different now. 
I assume the goal should be to have one "pipeline.py", which can eat zfits or hessio and knows how to treat any of them, right? But I assume that willl be done in a future step, correct? 

In that case I think this is fine here.

Reply:
Well, yes, the commented code is, as described at the top, just for testing compatibility between simtel and digicampipe pixel map, because I had some problems with it before. It is definitely just a temporary code and it will be surely removed but I would plead for keeping it for a while..

Oh, I see.. Sure, the final goal is to have one universal pipeline. But at this point, when we are still testing some settings specific for simulated data like baseline determination, I would say that separated pipe for simulations, which uses general functions and classes from digicampipe is less painful way.

Reply:
>  ... just for testing compatibility ...

This explanation virtually screams "make me a unit test... please!". :-D